From 9e33324644fff49b7aa15d34f836e72af8b32c78 Mon Sep 17 00:00:00 2001
From: Jerome Marchand <jmarchan@redhat.com>
Date: Tue, 11 Jun 2024 17:33:34 +0200
Subject: [PATCH] kernelshark: fix compiling error in LatencyPlot.cpp

Include the <algorithm> header to use std::for_each.

Fixes the following compilation error:
kernel-shark/src/plugins/LatencyPlot.cpp: In function void draw_latency(kshark_cpp_argv*, int, int, int):
kernel-shark/src/plugins/LatencyPlot.cpp:306:14: error: for_each is not a member of std
  306 |         std::for_each(range.first, range.second, lamPlotLat);
      |              ^~~~~~~~

Signed-off-by: Jerome Marchand <jmarchan@redhat.com>
---
 src/plugins/LatencyPlot.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/plugins/LatencyPlot.cpp b/src/plugins/LatencyPlot.cpp
index d41291745b77..5e3393ba7a36 100644
--- a/src/plugins/LatencyPlot.cpp
+++ b/src/plugins/LatencyPlot.cpp
@@ -15,6 +15,7 @@
 // C++
 #include <unordered_map>
 #include <iostream>
+#include <algorithm>
 
 // KernelShark
 #include "plugins/latency_plot.h"
-- 
2.47.0

